catalog/app/utils/Config.ts

```diff
@@ -0,0 +1,115 @@
+  alwaysRequiresAuth: boolean
+  analyticsBucket?: string
+  apiGatewayEndpoint: string
+  binaryApiGatewayEndpoint: string
+  calendlyLink?: string
+  googleClientId?: string
+  oktaClientId?: string
+  oktaBaseUrl?: string
+  intercomAppId?: string
+  mode: Mode
+  legacyPackagesRedirect?: string
+  linkedData?: {
+    name?: string
+    description?: string
+  }
+  mixpanelToken: string
+  noDownload?: boolean
+  noOverviewImages?: boolean
+  passwordAuth: AuthMethodConfig
+  registryUrl: string
+  s3Proxy: string
+  sentryDSN?: string
+  serviceBucket: string
+  ssoAuth: AuthMethodConfig
+  ssoProviders: string
+  build_version?: string // not sure where this comes from
+}
+
+const ajv = new Ajv({ allErrors: true, removeAdditional: true })
+
+ajv.addSchema(configSchema, 'Config')
+
+export class ConfigError extends BaseError {
+  static displayName = 'ConfigError'
+}
+
+const pprint = (x: unknown) => JSON.stringify(x, null, 2)
+
+function validateConfig(input: unknown): asserts input is ConfigJson {
+  if (ajv.validate('Config', input)) return
+  const printErrors = `Errors array:\n${pprint(ajv.errors)}`
+  const printInput = `Input:\n${pprint(input)}`
+  throw new ConfigError(
+    `invalid config format:\n${ajv.errorsText()}\n${printErrors}\n${printInput}`,
+    {
+      errors: ajv.errors,
+      input,
+    },
+  )
+}
+
+const AUTH_MAP = {
+  ENABLED: true,
+  DISABLED: false,
+  SIGN_IN_ONLY: 'SIGN_IN_ONLY',
+}
+
+const startWithOrigin = (s: string) => (s.startsWith('/') ? window.origin + s : s)
+
+const transformConfig = (cfg: ConfigJson) => ({
+  ...cfg,
+  passwordAuth: AUTH_MAP[cfg.passwordAuth],
+  ssoAuth: AUTH_MAP[cfg.ssoAuth],
+  ssoProviders: cfg.ssoProviders.length ? cfg.ssoProviders.split(' ') : [],
+  enableMarketingPages: cfg.mode === 'PRODUCT' || cfg.mode === 'MARKETING',
+  disableNavigator: cfg.mode === 'MARKETING',
+  s3Proxy: startWithOrigin(cfg.s3Proxy),
+  apiGatewayEndpoint: startWithOrigin(cfg.apiGatewayEndpoint),
+  binaryApiGatewayEndpoint: startWithOrigin(cfg.binaryApiGatewayEndpoint),
+  noDownload: !!cfg.noDownload,
+  noOverviewImages: !!cfg.noOverviewImages,
+  // XXX: there's no such field in the schema, so it will be stripped from the config when present
+  desktop: !!(cfg as any).desktop,
+})
+
+const getConfig = (input: unknown) => {
+  try {
+    validateConfig(input)
+    return transformConfig(input)
+  } catch (e) {
+    if (e instanceof ConfigError) throw e
+    throw new ConfigError('unexpected error', { originalError: e })
+  }
+}
+
+export type Config = ReturnType<typeof getConfig>
+
+const configKey = 'QUILT_CATALOG_CONFIG'
+const rawConfig = (window as any)[configKey]
+invariant(rawConfig, `window.${configKey} must be defined`)
+const CONFIG = getConfig(rawConfig)
+
+export default CONFIG
+
+export function useConfig(opts: { suspend: false }): { promise: Promise<Config> }
+export function useConfig(): Config
+/** @deprecated Config is now synchronous -- just import Config module directly */
```

Comment:
I would prefer to use getter function to get synchronous config

Reply:
It's quite important to easier refactoring

Reply:
And extending

Reply:
well, `useConfig` is a synchronous getter function, but it's a deprecated API anyways, bc doing `import cfg from 'utils/Config'` is more ergonomic

Reply:
`useConfig` is a hook, I don't like using it for config at all :)

---

```ts
import cfg from 'utils/Config'
const { noDownload } = cfg
```
has almost the same ergonomic as
```ts
import getConfig from `utils/Config`
const { noDownload } = getConfig()
```

But with `getConfig` you are free to extend it with additional functionality, like providing context or environment without much rewriting:
```ts
const { noDownload } = getConfig(user)
```

Reply:
I didn't get why 
```
import getConfig from `utils/Config`
const { noDownload } = getConfig()
```
should be like this:
```
import { getConfig } from 'utils/Config'
const cfg = getConfig()
const { noDownload } = cfg
```
but it's almost out of scope of this PR, because using getter it's not my hard requirement, more like my guts feeling

Reply:
> I didn't get why

because of this part:

> const { noDownload } = cfg

it seems quite meaningless to use destructuring here and it feels more convenient to access `cfg.*` members directly without assigning them to variables